addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js

```diff
@@ -19,7 +19,21 @@ export class ProductsWidgetControlPanel extends Component {
         this.notification = useService("pos_notification");
         this.orm = useService("orm");
         this.updateSearch = debounce(this.updateSearch, 100);
-        this.state = useState({ mobileSearchBarIsShown: false });
+        this.state = useState({ mobileSearchBarIsShown: false, isMobile: false });
+        this.productsWidgetControl = useRef("products-widget-control");
+
+        onMounted(() => {
+            this.toggleIsMobile();
+            window.addEventListener("resize", this.toggleIsMobile.bind(this));
+        });
+    }
+    toggleIsMobile() {
+        const element = this.productsWidgetControl.el;
+        if ((element && element.offsetWidth < 350) || this.env.isMobile) {
+            this.state.isMobile = true;
+        } else {
+            this.state.isMobile = false;
+        }
```

Comment:
Can't you just use the `ui_service` for this?

Basically, if you use that service, it will return you an object with `isSmall` getter. It's reactive so you don't need to worry about rerendering.

```js
setup() {
    this.ui = useService('ui');
}
```

And replace the use of `state.isMobile` with `ui.isSmall`.

Reply:
I added the service but i still need to use the width verification, I put a comment.